zstd-safe/zstd-sys/build.rs

```diff
@@ -215,26 +215,27 @@ fn compile_zstd() {
         .unwrap();
     #[cfg(feature = "zdict_builder")]
     fs::copy(src.join("zdict.h"), include.join("zdict.h")).unwrap();
-    cargo_print(&format!("cargo:root={}", dst.display()));
+    cargo_print(&format_args!("root={}", dst.display()));
 }
 
 /// Print a line for cargo.
 ///
 /// If non-cargo is set, do not print anything.
-fn cargo_print(content: &str) {
-    #[cfg(not(feature = "non-cargo"))]
-    println!("{content}");
+fn cargo_print(content: &dyn fmt::Display) {
```

Comment:
Could we even just take `content: impl fmt::Display` here? So we wouldn't need to add references to each usage.

Reply:
I suppose it could increase the build script's own build time though.

Reply:
In any case something we can look at later.